presto-main/src/test/java/io/prestosql/type/TestArrayOperators.java

```diff
@@ -749,6 +752,17 @@ public void testSubscript()
         assertDecimalFunction("ARRAY [2.22222222222222222, 2.3][1]", decimal("2.22222222222222222"));
     }
 
+    @Test
+    public void testObjectSubscript()
+    {
+        // Verify that timestamps with nanoseconds are represented as a non-slice object
+        assertTrue(TIMESTAMP_NANOS.getJavaType() == LongTimestamp.class && !Slice.class.isAssignableFrom(LongTimestamp.class));
```

Comment:
Looks very sturdy but then the expression below doesn't validate that the expression used is indeed a timestamp nanos, so the value of this check is comemntary

```suggestion
        // timestamp(9) has LongTimestamp stack representation (Object-based)
```

Reply:
I've removed that check and expanded the test slightly.